src/jit/lsra.cpp

```diff
@@ -6709,6 +6785,13 @@ void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool alloc
                     }
                     inVarToRegMap[assignedInterval->getVarIndex(compiler)] = REG_STK;
                 }
+#ifdef _TARGET_ARM_
+                // Consider overlapping floating point register for TYP_DOUBLE
+                else if (!assignedInterval->isConstant && isOverlappedRegRecord(assignedInterval, physRegRecord))
```

Comment:
>  where `assignedInterval` is `TYP_FLOAT`  but its `assignedReg` is not `physRegRecord`

If `assignedInterval` is `TYP_FLOAT`, then `isOverlappedRegRecord()` returns true only when `assignedReg` is `physRegRecord`, because there is a comparion at line 6556 which checks `assignedInterva` is `TYP_DOUBLE`.
```
6556 +    else if (assignedInterval->registerType == TYP_DOUBLE)
```

If you have meant "but its`assignedReg` is ~~not~~ `physRegRecord`", then you are definitely right.
However I think the first comparison in the if-statement at line 6779 already met and such case may not reach here.
```
6779    if (!assignedInterval->isConstant && assignedInterval->assignedReg == physRegRecord)
```
Are you suggesting narrowing down cases where `isOverlappedRegRecord()` returns `true` to avoid ambiguity ?
If so, I will rename and update the function, because the function is only used here right now.